### PR TITLE
Fix #8648 - Fixes to C3i in the ToE and implements Nova CEWs support in the ToE

### LINKED ISSUE #8648: [Issue] New C3i networks in Campaign cannot be created due to wrong condition
### Brief Description *

Image

Cannot create new C3i networks in MekHQ. However, the functionality already exists in Campaign.java. 

Seems to be an error in line 8893 in Campaign.java, in method `getAvailableC3iNetworks()` :

`if (en.hasC3i() && en.calculateFreeC3Nodes() < 5 && en.calculateFreeC3Nodes() > 0) {`

Likely an incorrect conditional in en.calculateFreeC3Nodes() < 5, tested with

`if (en.hasC3i() && en.calculateFreeC3Nodes() <= 5 && en.calculateFreeC3Nodes() > 0) {`

There's some oddities with how the networks display (e.g. it is unclear on the frontend which C3i computer belongs to what) but this seems to remedy the incapability at least, like so:

Image

Showing that MegaMek properly reads said C3i networks in MM:

Image

### 3. Steps to Reproduce

1. Field units with C3i computers.
2. Try to network them in MekHQ.

* Game may create network, but C3i units that are nodeless/unnetworked aren't being considered for possible choices for new "network." 

### Attach Files

[The Learning Ropes30470727_Constipation-part1.zip](https://github.com/user-attachments/files/24442055/The.Learning.Ropes30470727_Constipation-part1.zip)

### Severity *

Medium (Gameplay Limitation): Non-core functionality is impaired, providing a suboptimal but playable experience.

### MekHQ Suite Version *

0.50.11

### Operating System *

Windows 10

### Java Version *

Java Version 17.0.11 Microsoft

### Final Verification

- [x] I confirm this is a single, unique issue that hasn't been reported before
- [x] I have filled and provided all necessary information above
- [x] I have included any and all logs, custom units, and CPNX (save) files
- [x] I have asked on MegaMek Discord about this issue
- [x] I have confirmed this issue is being opened on the correct repository: MegaMek, MegaMekLab, or MekHQ

### PR DESCRIPTION
**Pull request by HammerGS**
Bug Fix: Issue #8648
              Issue #3414 

  Problem: C3i and NC3 networks could not be created because units with 5 free nodes (unnetworked units) were excluded from the available networks list.

  Root Cause: The condition calculateFreeC3Nodes() < 5 excluded unnetworked units, which have exactly 5 free nodes (since C3i/NC3 networks have a maximum of 6 nodes).

  Fix: Changed < 5 to <= 5 in:
  - Campaign.getAvailableC3iNetworks()
  - Campaign.getAvailableNC3Networks()

  New Feature: Nova CEWS Network Support

  Added full Nova CEWS network management to the TO&E context menu, following the existing C3i/NC3 patterns:

  Campaign.java:
  - Added getAvailableNovaCEWSNetworks() method
  - Added Nova CEWS handling in removeUnitFromForce() for proper network cleanup

  StaticChecks.java:
  - doAllUnitsHaveNovaCEWS() - checks if all selected units have Nova CEWS
  - areAllUnitsNotNovaCEWSNetworked() - checks if all units are unnetworked
  - areAllUnitsNovaCEWSNetworked() - checks if all units are networked
  - areAllUnitsOnSameNovaCEWSNetwork() - checks if all units share the same network

  TOEMouseAdapter.java:
  - Added Nova CEWS constants and action handler
  - Added context menu items: Create Network, Add to Network, Remove from Network, Disband Network

  Technical Notes

  - Nova CEWS has max 3 nodes (vs 6 for C3i/NC3), so unnetworked = 2 free nodes
  - Nova CEWS shares UUID infrastructure with NC3 (setNC3NextUUIDAsString)
  - Network ID prefix: C3Nova.

  Test Plan

  Added two new test files with 34 tests total:

  - C3NetworkTest.java - 14 tests covering:
    - C3i networks include unnetworked units (5 free nodes) - regression test for #8648
    - C3i networks include partially filled networks
    - C3i networks exclude full networks (0 free nodes)
    - C3i networks exclude units not in TO&E
    - C3i networks exclude non-C3i units
    - C3i networks return unique entries only
    - NC3 networks include unnetworked units - regression test for #8648
    - NC3 networks exclude full networks
    - Nova CEWS networks include unnetworked units (2 free nodes)
    - Nova CEWS networks include partially filled networks
    - Nova CEWS networks exclude full networks
    - Nova CEWS networks exclude units not in TO&E
    - Nova CEWS networks exclude non-Nova units
    - Nova CEWS networks exclude invalid free node counts
  - StaticChecksNovaCEWSTest.java - 20 tests covering all four helper methods with various edge cases including null entities and empty vectors

  Files Changed

  | File                          | Changes                             |
  |-------------------------------|-------------------------------------|
  | Campaign.java                 | +44 lines (fix + Nova CEWS methods) |
  | TOEMouseAdapter.java          | +73 lines (Nova CEWS menu handling) |
  | StaticChecks.java             | +57 lines (4 helper methods)        |
  | C3NetworkTest.java            | +360 lines (new file)               |
  | StaticChecksNovaCEWSTest.java | +292 lines (new file)               |

  ---
  Fixes #8648

### CODE REVIEW

File: MekHQ/src/mekhq/gui/adapter/TOEMouseAdapter.java [hunk 1]
```diff
@@ -1473,6 +1495,57 @@
...
+                            menuItem.setActionCommand(TOEMouseAdapter.COMMAND_ADD_TO_NETWORK +
+                                                            network[0] +
+                                                            '|' +
+                                                            unitIds);
+                            menuItem.addActionListener(this);
+                            menuItem.setEnabled(true);
+                            availMenu.add(menuItem);
+                        }
+                    }
+                    JMenuHelpers.addMenuIfNonEmpty(networkMenu, availMenu);
```
[the remark this answers is not in the document]
Comment (HammerGS) [reply MekHQ/src/mekhq/gui/adapter/TOEMouseAdapter.java]: Addressed https://github.com/MegaMek/mekhq/pull/8649/commits/2ac214ce6b0eb5e81e6b7334e9483b1a03d4f8e4

[the remark this answers is not in the document]
Comment (HammerGS) [reply MekHQ/src/mekhq/gui/adapter/TOEMouseAdapter.java]: Addressed https://github.com/MegaMek/mekhq/pull/8649/commits/2ac214ce6b0eb5e81e6b7334e9483b1a03d4f8e4
